tagcompare/logger.py

```diff
@@ -0,0 +1,56 @@
+import os
+import logging
+
+import settings
+
+
+class Logger(object):
+    def __init__(self, name, directory=None, writefile=False):
+        if directory and not writefile:
+            raise ValueError("directory was specified when writefile is False!")
+
+        self.__directory = directory
+        if writefile:
+            if not directory:
+                self.__directory = os.path.join(
+                    os.path.abspath(os.path.dirname(__file__)), settings.LOG_DIR)
+            if not os.path.exists(self.__directory):
+                raise IOError("directory doesn't exist at {}".format(self.__directory))
```

Comment:
This makes an error for me:

```python
$ py.test tagcompare/test/test_logger.py 
============================================================ test session starts ============================================================
platform linux2 -- Python 2.7.9 -- py-1.4.26 -- pytest-2.6.4
collected 4 items 

tagcompare/test/test_logger.py .F..

================================================================= FAILURES ==================================================================
_________________________________________________________ test_init_with_writefile __________________________________________________________

    def test_init_with_writefile():
>       testlogger = logger.Logger(name="test2", writefile=True)

tagcompare/test/test_logger.py:14: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <tagcompare.logger.Logger object at 0x7f5251ae4a90>, name = 'test2', directory = None, writefile = True

    def __init__(self, name, directory=None, writefile=False):
        if directory and not writefile:
            raise ValueError("directory was specified when writefile is False!")
    
        self.__directory = directory
        if writefile:
            if not directory:
                self.__directory = os.path.join(
                    os.path.abspath(os.path.dirname(__file__)), settings.LOG_DIR)
            if not os.path.exists(self.__directory):
>               raise IOError("directory doesn't exist at {}".format(self.__directory))
E               IOError: directory doesn't exist at /home/phil/Documents/tagcompare/tagcompare/output

tagcompare/logger.py:18: IOError
==================================================== 1 failed, 3 passed in 0.01 seconds =====================================================
```

Probably because I didn't do `make install`, which creates the output directory? Probably the tests for the logger shouldn't depend on that.

Only tangentially related, but probably nothing should be writing to directories inside the python package, either. Python packages tend to get installed under something like `/usr/lib`, and people don't expect mutable application data there.